### PR TITLE
runtime: Patch gitconfig.vim syntax

### PR DESCRIPTION
**Pull request by adrian5**
Related files that are already in sync with vim:

* runtime/ftplugin/gitconfig.vim
* runtime/indent/gitconfig.vim

### CODE REVIEW

File: runtime/syntax/gitconfig.vim [hunk 1]
```diff
@@ -2,24 +2,22 @@
...
   finish
 endif
 
-setlocal iskeyword+=-
-setlocal iskeyword-=_
 syn case ignore
 syn sync minlines=10
 
-syn match   gitconfigComment	"[#;].*"
+syn match   gitconfigComment	"[#;].*" contains=@Spell
```
Comment (janlazo) [inline runtime/syntax/gitconfig.vim:14]: That did not come from https://github.com/vim/vim/commit/c08ee7476b19f9b4de5df287797af87c4e3fba0a.

Comment (adrian5) [reply runtime/syntax/gitconfig.vim:14]: Oh, you're right – that was [5be4cee](https://github.com/vim/vim/commit/5be4ceecea5520265066eac972460ebb1cdf05e7#diff-8389194b233664291c48f8ffae3321641df25f459d121473b552fb803d16c538). Should I add that to the commit message before c08ee74?

Comment (janlazo) [reply runtime/syntax/gitconfig.vim:14]: Yes. Please split them in separate commits like how Vim did it. Each ported Vim commit should match 1-1 to the original Vim commit with respect to the modified files.

Comment (adrian5) [reply runtime/syntax/gitconfig.vim:14]: I see, that makes sense. I'll have to check my other PR #13333 for the same issue. So don't waste any time on that now, I'll add a note when I have checked it. Thanks.

**Pull request closed without merging**